hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/file/OMFileRequest.java

```diff
@@ -607,11 +609,20 @@ public static OzoneFileStatus getOMKeyInfoIfExists(
       OMMetadataManager omMetadataMgr, String volumeName, String bucketName,
       String keyName, long scmBlockSize) throws IOException {
 
+    OMFileRequest.validateBucket(omMetadataMgr, volumeName, bucketName);
+
     Path keyPath = Paths.get(keyName);
     Iterator<Path> elements = keyPath.iterator();
     String bucketKey = omMetadataMgr.getBucketKey(volumeName, bucketName);
     OmBucketInfo omBucketInfo =
             omMetadataMgr.getBucketTable().get(bucketKey);
+    if (null == omBucketInfo) {
```

Comment:
Does validateBucket handle the non-exist bucket case?

Reply:
OK, will remove this. Its a duplicate check.